UPGRADE-API-1.12.md

```diff
@@ -0,0 +1,58 @@
+# UPGRADE FROM `v1.11.X` TO `v1.12.0`
+
+1. The `Sylius\Bundle\ApiBundle\DataProvider\CartShippingMethodsSubresourceDataProvider` has been changed to `Sylius\Bundle\ApiBundle\DataProvider\ShippingMethodsCollectionDataProvider`.
+
+1. The `Sylius/Bundle/ApiBundle/Serializer/ShippingMethodNormalizer` logic and constructor has been changed due to refactor above.
+
+    ```diff
+        public function __construct(
+            private OrderRepositoryInterface $orderRepository,
+            private ShipmentRepositoryInterface $shipmentRepository,
+            private ServiceRegistryInterface $shippingCalculators,
+    +       private RequestStack $requestStack,
+    +       private ChannelContextInterface $channelContext
+        ) {
+            ...
+        }
+    ```
+   
+1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise causes empty array `[]` in response and correct parameters returns shipping methods available for your `shipment`. Here is how the response look:
```

Comment:
```suggestion
1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise cause empty array `[]` in response and correct parameters return shipping methods available for your `shipment`. Here is what the response looks like:
```
And divide this line into several lines